### PR TITLE
Bump puma to 3.12.2

### PR DESCRIPTION
**Pull request by abellotti**
- taking care of a Hakiri warning.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ gem 'manageiq-password',   '~> 0.2', ">= 0.2.1"
 gem 'mimemagic',           '~> 0.3.3'
 gem 'more_core_extensions'
 gem 'pg',                  '~> 1.0', :require => false
-gem 'puma',                '~> 3.0'
```
Comment (bdunne) [inline Gemfile]: Do we need to lock down to 3.12?  Is `~>3.0, >=3.12.2` ok?

Comment (abellotti) [reply Gemfile]: I think that would work too.  was trying to be consistent with sources.

Comment (abellotti) [reply Gemfile]: Somehow Hakiri did not like ~>3.0, >=3.12.2,  moving to just >= 3.12.2 worked.

Comment (agrare) [reply Gemfile]: If I remember right hakiri wanted `>= 3.12.2` first, then `~> 3.0`

Comment (abellotti) [reply Gemfile]: Ahh, lemme try that.

Comment (agrare) [reply Gemfile]: Wow really can't make hakiri happy...

Comment (abellotti) [reply Gemfile]: holy cow, REALLY picky, tried '>= 3.12.2' with '~> 3.0', then '~> 3.12', and finally with '~> 3.12.2', still failed.  looks like the only one that works is the original one like sources, i.e. just '~> 3.12.2'

**Comment by bdunne** [comment]
Any reason we shouldn't upgrade to v4? https://github.com/RedHatInsights/topological_inventory-api/pull/276

**Comment by bdunne** [comment]
Closing in favor of #276

**Pull request closed without merging**